File: code_combat/4_Backwoods_Forest/152-Go_Fetch/go_fetch.py
def goFetch():
    while True:
        potion = hero.findNearestItem()
        if potion:
            pet.fetch(potion)

pet.on("spawn", goFetch);

while True:
    if hero.canCast("haste"):
        hero.cast("haste", pet)
    elif hero.isReady("reset-cooldown"):
        hero.resetCooldown("haste")

File: projects/code_combat/4_Backwoods_Forest/152-Go_Fetch/goFetch.js
// You've been caught in a burl trap!
// Send your pet to fetch the health potions!

function goFetch() {
    // You can use loops in a handler function.
    while(true) {
        var potion = hero.findNearestItem();
        if(potion) {
            // Use pet.fetch to fetch the potion.
            pet.fetch(potion);
        }
    }
}

// When your pet is summoned, it triggers a "spawn" event.
// This tells your pet to run the goFetch function at the start of the level:
pet.on("spawn", goFetch);
